### ISSUE TITLE
Node: Nested payload in TransactionDataPayload is not properly typed

### ISSUE THREAD

**Issue by kwek20** [iotaledger/iota-sdk#1249]
### Issue description

parsing nested TransactionDataPayload causes class-transformer to stop due to cyclic parsing.

resulting in 
```
Block data: Block {
  protocolVersion: 2,
  parents: [
    '0x3689742bb257ffec5b9789022bb0af90c23574b605d208b84d2abd083d54085d'
  ],
  payload: {
    essence: {
      type: 1,
      networkId: '1856588631910923207',
      inputs: [Array],
      inputsCommitment: '0xff6bd7f6aeb9c6926da84267d859cf08ce44c228c13fb67b8893d642030a2f51',
      outputs: [Array],
      payload: [Object]
    },
    unlocks: [ [Object] ]
  },
  nonce: '233605'
} 
```

instead of 
```
Block data: Block {
payload: TransactionPayload {
    type: 6,
    essence: RegularTransactionEssence {
      type: 1,
      networkId: '1856588631910923207',
      inputsCommitment: '0xff6bd7f6aeb9c6926da84267d859cf08ce44c228c13fb67b8893d642030a2f51',
      inputs: [Array],
      outputs: [Array],
      payload: [Object]
    },
    unlocks: [ [SignatureUnlock] ]
  },
  nonce: '233605'
} 
```

possible fix: Custom @transform with a new `plainToInstance` call like here:
https://github.com/typestack/class-transformer/issues/212#issuecomment-451066261

### Version

1.1

### Expected behaviour

nested payload has a type

### Actual behaviour

no type

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue

1. Change client 07 example to look up a transaction with payload type 6
2. yarn run-example ./client/07-get-block-data.ts
3. see that block data isnt typed

### Duplicate declaration

- [X] I have searched the issues tracker this issue and there is none

**Comment by kwek20** [comment]
not fixed yet in https://github.com/iotaledger/iota-sdk/pull/1253, just solved.
Needs a better solution for future different Payloads inside Transaction

**Comment by msarcev** [comment]
Maybe this can be a temporary fix while we find a better one. At the moment it would be nice to have a fix to unblock the intergration in Explorer.
